mvc/overview.rst

```diff
@@ -1,8 +1,224 @@
+
+Tag Helpers
+-----------
+:doc:`Tag Helpers </views/tag-helpers/intro>` enable you to dynamically generate or modify the HTML returned to the client. You can use tag helpers to define custom tags or to modify the behavior existing tags. Tag Helpers bind to specific elements based on the element name and its attributes. Tag Helpers give you all of the benefits of server-side rendering while still preserving an HTML editing experience.
+
+For example, you can use built-in ``LinkTagHelper`` to create a link to the ``Login`` action of your ``AccountsController`` like this:
+
+.. code-block:: html
+  :emphasize-lines: 3
+  
+  <p>
+      Thank you for confirming your email. 
+      Please <a asp-controller="Account" asp-action="Login">Click here to Log in</a>.
+  </p>
+
+Or you can use the ``EnvironmentTagHelper`` to include different scripts in your views (ex full vs minified) based on the runtime environment (i.e. Development vs Production):
+
+.. code-block:: html
+
+  <environment names="Development">
+      <script src="~/lib/jquery/dist/jquery.js"></script>
+  </environment>
+  <environment names="Staging,Production">
+      <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.1.4.min.js"
+              asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
+              asp-fallback-test="window.jQuery">
+      </script>
+  </environment>
+
+MVC includes a full set of built-in Tag Helpers that you can use or your can build your own.
+
+View Components
+---------------
+:doc:`View Components </views/view-components>` are like partial views with associated application buisiness logic. View Components can have asynchronous logic and can be reused throughout your application views.
+
+Dependency Injection
+--------------------
+
+MVC is built on the :ref:`dependency injection (DI) <aspnet:dependency-injection>` support in ASP.NET. Support for DI is plumbed throughout MVC. You can inject services into controllers, models, views, filters and View Components.
+
+Filters
+-------
+
+:doc:`Filters </controllers/filters>` enable you to specify pre and post processing logic for control action methods. Use filters to handle cross-cutting concerns, like authorization and exception handling, without having to repeat the same logic across all of your controller actions.
+
+Areas
+-----
+
+:doc:`Areas </controllers/areas>` provides a way to separate a large MVC application into semantically-related groups of models, views, and controllers.
+
+Mobile Development
+------------------
+
+MVC has great support for :doc:`mobile development </views/mobile>`, including the ability to create mobile-specific views to give your customers the best possible experience on their devices.
+
+Web UI and Web APIs
+-------------------
+
+In addition to being a great platform for build websites MVC has great support for building Web APIs. With MVC you can build RESTful services that can reach a broad ranch of clients including browsers and mobile devices. 
```

Comment:
Line 201 above, should it not be "...platform for building websites, MVC has great..."?